### ISSUE TITLE
macOS: Do you want the "???.app" to accept incoming network connections?

### ISSUE THREAD

**Issue by Hanterdro** [nwjs/nw.js#6596]
nwjs Version : 0.30.0 - 0.30.2
Operating System : macOS 10.13.4

### Expected behavior

Don't get the a message dialog about incoming network connections.

### Actual behavior

After building a nwjs app I get since version 0.30.0+ the gatekeeper dialog:

```
Do you want the "???.app" to accept incoming network connections?
```

### How to reproduce

Build a nwjs app on macOS with `nwjs-builder-phoenix` 1.14.6:

```
build ---tasks mac-x64 --mirror https://dl.nwjs.io/ .
```

package.json

```
  "build": {
    "nwVersion": "0.30.0",
  },
```

nwjs prior to 0.30.0 don't trigger the incoming network connection dialog.

**Comment by wvary** [comment]
I see this issue with my build.
I'm using SDK v0.30.2.
I'm on MacOS Sierra.
Node version is 9.11.1.
grunt-nw-builder v2.0.0

Thanks.

**Comment by wvary** [comment]
I've created a repo to show how we reproduce this issue.
I also see the prompt using nwjs-builder-phoenix.

Here's the repo: https://github.com/wvary/nwjs-firewall-test

1. Build with nwbuild
```
git clone https://github.com/wvary/nwjs-firewall-test.git
npm run build

```

2. Build with nwjs-builder-phoenix
```
git checkout try-nwjs-builder-phoenix
npm install
npm run build 
```

Both builds show the firewall prompt.

**Comment by Christywl** [comment]
I still can't reproduce it with above steps. Are there some special settings for the firewall in your Mac?

**Comment by Hanterdro** [comment]
not for me

**Comment by wvary** [comment]
Does anyone know if there's a version of nwjs that doesn't show this prompt?
Also, is the issue with nwjs or the build process?

**Comment by Hanterdro** [comment]
nwjs prior to 0.30 does not have this issue

**Comment by wvary** [comment]
Thanks @Hanterdro , version 0.29.4 does not show the prompt.
I'll use that version until the issue is fixed.

**Comment by Hanterdro** [comment]
@wvary Could you maybe upload into your test repo a mac build with 0.30.x? If @Christywl would get with these uploaded build the same error as we get we could isolate the issue maybe.

**Comment by wvary** [comment]
@Hanterdro @Christywl 

I've uploaded a dist.tar file to my google drive.  It's shared as public so please take a took.
https://drive.google.com/file/d/1T-NZMrERAsWqLtLkYOFkr7u6rWq0V02G/view?usp=sharing

Thanks,

Willet

**Comment by Christywl** [comment]
I still can't reproduce this issue with the uploaded dist.tar file on my Mac. I try to set the firewall in System Preferences, the app works fine when the firewall is on or off.
<img width="1280" alt="screen shot 2018-05-10 at 09 22 58" src="https://user-images.githubusercontent.com/20039821/39847366-c6e5d19c-5433-11e8-9456-3b4b24743419.png">

@rogerwang , could you please help confirm this in your Mac? It works fine in my Mac(10.12.6 and 10.13.3).

**Comment by Hanterdro** [comment]
@Christywl
I tried @wvary a few days ago on my companies Mac without any trouble.

A few minutes ago I tested the same file on my private Mac.

1) I had to disable temporary gatekeeper to start the file
`sudo spctl --master-disable` `sudo spctl --master-enable`
Because gatekeeper was yelling that the file was corrupt and should be deleted...

2) After starting the test app without gatekeeper I got the network connection alert again.

On my private macOS I am using `10.13.4`. I think on my companies Mac is 10.13.3 (I will verify this tomorrow).

Maybe it is an  macOS  `10.13.4` in combination with nwjs `0.30.0`+ issue?

**Comment by Christywl** [comment]
I try the sample on Mac 10.13.4, and don't get this alert. The sample also works fine.

**Comment by Hanterdro** [comment]
@Christywl 

I think I may have found something. Please go to your *Firewall Options* in *Security & Privacy* -> *Firewall* and delete all entries of nwjs-firewall-test **and** nodejs. After I deleted the nodejs firewall settings I can reproduce the issue with @wvary  binary.

**Comment by Hanterdro** [comment]
Looks like that nwjs 0.31.0 fixed this issue for me. Can you confirm this @wvary ?

**Comment by wvary** [comment]
On the surface it appears that 0.31.0 has fixed the issue, but as soon as I removed the entry from firewall options, then the dialog appears again.  The difference with this version vs. the 0.30 is that 0.30 will show the dialog regardless of the firewall options entries, while 0.31.0 will only show the dialog once.  Once the entry appears in the firewall options, the dialog stops showing.

To show the dialog again, simply delete the entry and the dialog appears again.

To recap, version 0.29.4 will not show the dialog even if the entry is not in the firewall options.

In summary, the issue isn't totally fixed IMO.

**Comment by indexofrefraction** [comment]
this problem is **not solved** at all
same issue in stable 0.37.0 on macos 10.12.6

**Comment by marc7000** [comment]
I have the same issue with version 0.38.4 under macOS 10.13.6. 
On every start of the app the popup appears and asks for permission for incoming network connections, no matter if the setting has already been set in the firewall.
![Bildschirmfoto 2019-07-22 um 09 40 01](https://user-images.githubusercontent.com/1118837/61615410-3b1e6480-ac66-11e9-9079-47b6b6f99f60.png)
